src/common/util.js

```diff
@@ -238,7 +240,13 @@ function hasSpaces(text, index, opts) {
   return idx !== index;
 }
 
-function locStart(node) {
+function locStart(node, locStartFromPlugin) {
```

Comment:
I think we should delete `locStart` and `locEnd` from `util.js` and force each parser (in plugins) to provide it. This way we won't have to have awkward code like `util.locEnd(node, options.locEnd);`.

Reply:
I agree, that would be much cleaner. It does require many changes though, because every callsite has to be changed (see https://github.com/prettier/prettier/pull/3777/commits/b2937f87d467eb97f57ba5fad065b41eb0cfc839 some first progress on this). I don't mind doing those changes, but let me know if you'd implement this differently!

Reply:
Looking good to me 👍

Reply:
Awesome, thanks 😃